src/shogun/regression/LeastAngleRegression.cpp

```diff
@@ -316,6 +316,9 @@ bool CLeastAngleRegression::train_machine_templated(CDenseFeatures<ST>* data)
 					map_Xa.block(0, i_kick+1, numRows, numCols-i_kick).eval();			
 		}
 
+		// Observe this value directly
+		observe(ObservedValue::make_observation(nloop, std::string("betas"), make_any(beta)));
```

Comment:
The issue I'm facing here is how do we explain/show to the user that he can observe this value `betas`? I thought that we could register somehow some "fake" parameters inside the class which cannot be updated/obtained with `put()`/`get()`, but they can only be observed by the user (e.g. `SG_ADD(&beta, "betas", "Weights of the model while training", ParameterProperties::OBSERVABLE);`). @karlnapf what do you think?

Reply:
Keep in mind registering parameters happens at *runtime*....so in fact you could only register the parameter in the observer once you receive it? Kinda crazy but might work :) Might require some changes. Let me know if you see what I mean

Reply:
I was thinking more about registering inside the machine directly the parameter we are going to observe. For LARS it could be something like:
```c++
void CLeastAngleRegression::init()
{
	m_lasso = true;
	m_max_nonz = 0;
	m_max_l1_norm = 0;
	m_epsilon = CMath::MACHINE_EPSILON;
	SG_ADD(&m_epsilon, "epsilon", "Epsilon for early stopping", ParameterProperties::HYPER);
	SG_ADD(&m_max_nonz, "max_nonz", "Max number of non-zero variables", ParameterProperties::HYPER);
	SG_ADD(&m_max_l1_norm, "max_l1_norm", "Max l1-norm of estimator", ParameterProperties::HYPER);
	SG_ADD(&m_lasso, "lasso", "Max l1-norm of estimator", ParameterProperties::HYPER);

        SG_ADD(&beta, "betas", "Weights of the model while training", ParameterProperties::OBSERVABLE);
}
```
The `betas` parameter will not be modifiable by `put()`\`get()` and it can only be captured using an observer. Therefore, the user may be able to see which kind of information can be extracted from the model at runtime using an observer. 

Obviously, I do not know if SG_ADD is the right method, but I would like to signal somehow the presence of these emitted values.

See what I mean?